test/powershell/Language/Classes/Scripting.Classes.Modules.Tests.ps1

```diff
@@ -71,3 +71,41 @@ Import-Module Random
     }
 
 }
+
+Describe 'Module reloading with Class definition' -Tags "CI" {
+
+    BeforeAll {
+        Set-Content -Path TestDrive:\TestModule.psm1 -Value @'

```

Comment:
@daxian-dbw I couldn't reproduce your last [post](https://github.com/PowerShell/PowerShell/pull/2837#discussion_r90740266).  I wrote [another test](https://github.com/alx9r/ToolFoundations/blob/6d8125f84f03ffe1ba5b0918f199b8ef63870db9/EnvironmentTests/dynamicModulesWithClassesTests.ps1) that creates a dynamic module with arguments, then creates (overwrites?) another identical module with different arguments, then creates another with a slightly different class definition.

The results are the same with PowerShell 5.0, 5.1, and 6.0.0-alpha.13:

 * fresh module works as expected
 * creating new module that differs only by module arguments results in stale data returned by class instance
 * creating new module whose scriptblock differs slightly works as expected

This behavior seems very similar to how I understand reloading of file-backed modules behaves.